webextension/background.js

```diff
@@ -1,3 +1,174 @@
+const assignManager = {
+  closeableWindows: new Set([
+    "about:startpage",
+    "about:newtab",
+    "about:home",
+    "about:blank"
+  ]),
+  menuAssignId: "open-in-this-container",
+  menuRemoveId: "remove-open-in-this-container",
+  storageArea: browser.storage.local,
+
+  init() {
+    browser.tabs.onActivated.addListener((info) => {
+      browser.tabs.get(info.tabId).then((tab) => {
+        this.calculateContextMenu(tab);
+      }).catch((e) => {
+        throw e;
+      });
+    });
+
+    browser.windows.onFocusChanged.addListener((id) => {
+      browser.tabs.query({active: true}).then((tabs) => {
+        if (tabs && tabs[0]) {
+          this.calculateContextMenu(tabs[0]);
+        }
+      }).catch((e) => {
+        throw e;
+      });
+    });
+
+    browser.runtime.onMessage.addListener((request) => {
```

Comment:
Can we change `request` to something more accurate like `neverAskMessage`? When I first read this, I thought `request` was an HTTP request of some kind.